pyuvdata/mwa_corr_fits.py

```diff
@@ -69,9 +69,78 @@ def correct_cable_length(self, cable_lens):
 
+    def flag_init(self, num_fine_chan, edge_width=80e3, start_flag=2.0,
+                  end_flag=2.0, flag_dc_offset=True):
+        """
+        Do routine flagging of the edges, beginning and end of obs, as well as
+        the center fine channel of each coarse channel.
+
+        Parameters
+        ----------
+        edge_width: float
+            The width to flag on the edge of each coarse channel, in hz. Set to
+            0 for no edge flagging.
+        start_flag: float
+            The number of seconds to flag at the beginning of the observation.
+            Set to 0 for no flagging.
+        end_flag: floats
+            The number of seconds to flag at the end of the observation. Set to
+            0 for no flagging.
+        flag_dc_offset: bool
+            Set to True to flag the center fine channel of each coarse channel.
+
+        Raises
+        ------
+        ValueError
+            If edge_width is not an integer multiple of the channel_width of the data (0 also acceptable).
+            If start_flag is not an integer multiple of the integration time (0 also acceptable).
+            If end_flag is not an integer multiple of the integration time (0 also acceptable).
+        """
+        if (edge_width % self.channel_width) > 0:
+            raise ValueError("The edge_width must be an integer multiple of the"
+                             "channel_width of the data or zero.")
+        if (start_flag % self.integration_time[0]) > 0:
+            raise ValueError("The start_flag must be an integer multiple of the"
+                             "integration_time of the data or zero.")
+        if (end_flag % self.integration_time[0]) > 0:
+            raise ValueError("The end_flag must be an integer multiple of the"
+                             "integration_time of the data or zero.")
+
+        num_ch_flag = int(edge_width / self.channel_width)
+        num_start_flag = int(start_flag / self.integration_time[0])
+        num_end_flag = int(end_flag / self.integration_time[0])
+
+        if num_ch_flag > 0:
+            edge_inds = []
+            for ch_count in range(num_ch_flag):
+                # count up from the left
+                left_chans = list(range(ch_count, self.Nfreqs, num_fine_chan))
+                # count down from the right
+                right_chans = list(range(self.Nfreqs - 1 - ch_count, 0, -num_fine_chan))
+                edge_inds = edge_inds + left_chans + right_chans
+
+            self.flag_array[:, :, edge_inds, :] = True
+
+        if flag_dc_offset:
+            center_inds = list(range(num_fine_chan // 2, self.Nfreqs, num_fine_chan))
+
+            self.flag_array[:, :, center_inds, :] = True
+
+        if (num_start_flag > 0) or (num_end_flag > 0):
+            shape = self.flag_array.shape
+            reshape = [self.Ntimes, self.Nbls, self.Nspws, self.Nfreqs, self.Npols]
+            self.flag_array = np.reshape(self.flag_array, reshape)
```

Comment:
This is really requiring a particular ordering of the baseline-time axis, which is not in general guaranteed for UVData objects. I guess it's ok if this is only ever called when the data are initially read in and that reading guarantees an ordering (need to check that with @PyxieLouStar). It can be written more generally by making a list of the time stamps, picking out the ones you want to flag and then finding their position in the `time_array`.

Reply:
I believe this shape is required on read - conferred with Pyxie on this one. You can see it is also used on lines 565-567. I see you've tagged her so hopefully she'll chime in here as well.

Reply:
I forced time as the primary axis and baseline as the secondary axis when the time and baseline axes get combined after reading in the data for no real reason except that ordering made more sense to me.

Reply:
Actually, I set it up that way because that's how a cotter file is organized when pyuvdata reads it in.

Reply:
ok. And the idea is that this function would only ever get called on read?

Reply:
This particular function would only get called on read, although now that I think about it, it might have been nice to write the function in utils so it could be called whenever. A complication that comes to mind in that scenario is that if one wants to call it after averaging then they will need to be careful with the flag_dc_offset keyword.